lib/dynamoid/criteria/chain.rb

```diff
@@ -140,13 +140,13 @@ def records
           records_via_query
         else
           records_via_scan
-        end
+        end.lazy.flat_map(&:itself)
```

Comment:
`records_via_query `/`records_via_scan` mean "records" not "pages".

So it's reasonable either to make them return lazy collection of hashed (like they did before) or rename them to `pages_via_query`/`pages_via_scan`

Reply:
Sounds good, this afternoon is gonna be moved to fixing stuff like that and creating the final paged public method. Yesterday was just getting my hooks into it all and specs green again.

Just didn't want to get too far before showing this since it is a fairly fundamental change in internal datagram